evalml/pipelines/components/transformers/imputers/target_imputer.py

```diff
@@ -93,9 +99,9 @@ def fit(self, X, y):
             raise TypeError("Provided target full of nulls.")
         y = y.to_frame()
 
-        # Convert all bool dtypes to category for fitting
+        # Return early since bool dtype doesn't support nans and sklearn errors if all cols are bool
```

Comment:
Coming off of the discussion in the simple imputer, this raises alarm bells for me. I don't think it's actually wrong, we only need the target imputer during pipeline fit, so if the training data is bool with no nans this component is a no-op. I think it would be better here to state the logical reason for why we don't need to fit on a bool column (akin to what I said above) than the technical reason, since a comment like this could be confusing in the future (it sounds like there's a scikit bug we'd have to revisit, which isn't the case)

Reply:
Good call - I'll update the comment to just say we're returning early because there would never be any null values. In the SimpleImputer, I did end up deciding to mention the sklearn behavior when we convert to `boolean`, because that behavior is explicitly to work around that sklearn error, but I do also mention that we need to fit the `_component_obj` then in case the dtype is different in the training data.